v22.1/security-reference/scram-authentication.md

```diff
@@ -0,0 +1,323 @@
+
+Allow *only* `scram-sha-256` (not the weaker `password`) in your [Authentication Configuration (HBA)](authentication.html#authentication-configuration).
+
+{{site.data.alerts.callout_danger}}
+After this change, users will not be able to sign in with plaintext passwords. All critical SQL users must be migrated to SCRAM-SHA-256 authentication before making this change.
+{{site.data.alerts.end}}
+
+{% include_cached copy-clipboard.html %}
+```sql
+SET CLUSTER SETTING server.host_based_authentication.configuration TO '
+# TYPE    DATABASE  USER          ADDRESS     METHOD
+# any other higher-priority rules ...
+  host    all       human         all         scram-sha-256
+  host    all       app_user      all         cert
+  host    all       cyborg        all         cert-scram-sha-256
+# any other lower-priority rules ...
+```
+
+### Secure handling of plaintext credentials
+
+Using SCRAM authentication, a CockroachdDB cluster need not store or handle the plaintext passwords used by clients to authenticate. When managing SQL users and roles, only salt and hash(salt, password) need be stored.
+
+However, to achieve full separation of concerns, the plaintext credentials must be isolated to the extent possible: they must be exposed as little as possible in transit, storage and during computation.
+
+Generally, this can be best achieved by handling plaintext credentials only in a dedicated credential management environment, such as a secure compute instance or secure physical workstation.
+
+That environment must be a provisioned with:
+
+- A tool for [creating PostgreSQL/CockroachDB-formated SCRAM-SHA-256 hashes](#managing-users-with-pre-hashed-passwords), such as the example script provided in the [appendix](#appendix-python-scram-hashing-script).
+- A CRDB client with ['USER CREATE/ALTER privileges'](../create-user.html#create-a-user-that-can-create-other-users-and-manage-authentication-methods-for-the-new-users)
+- Access, with write privileges, to a secrets store where the plaintext passwords will be persisted. Recommended solutions include Hashicorp Vault, GCP secrets manager or AWS secrets manager.
+
+On that instance, [generate SCRAM salted password hashes](#managing-users-with-pre-hashed-passwords), manage the CRDB users via CRDB client and persist the secrets in the secrets store, as described in the following section.
+
+## Managing users with pre-hashed passwords
+
+[Many database tools](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.scram.html) support SCRAM-SHA-256 and can be used to calculate the hashes servers will need for authentication.
+
+In this example, we'll use this [Python script](#appendix-python-scram-hashing-script), which, like many good things, comes from the [open-source community](https://stackoverflow.com/questions/68400120/how-to-generate-scram-sha-256-to-create-postgres-13-user).
+
+### Start a python shell
+
+For example, with docker.
+
+{% include_cached copy-clipboard.html %}
+```bash
```

Comment:
Hmmm, I thought `bash` would work for syntax here, but it's not showing the prompt rendered, so switch out for `shell`